platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py

```diff
@@ -748,6 +765,14 @@ def get_reboot_cause(self):
             to pass a description of the reboot cause.
         """
         #read reboot causes files in the following order
+
+        # To avoid the leftover hardware reboot cause confusing the reboot cause determine service
+        # Skip the hardware reboot cause check if warm/fast reboot cause found from cmdline
+        if utils.is_host():
+            reboot_cause = self._parse_warmfast_reboot_from_proc_cmdline()
```

Comment:
reboot_cause will be None when system is booting after user issued "sudo reboot". ?

Reply:
New function _parse_warmfast_reboot_from_proc_cmdline() is designed to check the /proc/cmdline to see whether the last boot is a warm/fast reboot,  if Not, then the local variable 'reboot_cause' will be 'None', but this 'reboot_cause' is not the final reboot cause which will be returned by this platform API in this case, there will be further logic to determine the reboot cause from the sysfs if you check the reset of the code.